RetailCoder.VBE/Refactorings/Rename/RenameEventHandler.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Rubberduck.Common;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.UI;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Rubberduck.Refactorings.Rename
+{
+    public class RenameEventHandler : RenameHandlerBase
```

Comment:
> It really feels like each implementation should actually be a specialized IRefactoring implementation

 After considering the above approach for the last couple days, it turns out that I'd like to lobby for keeping the Renaming classes free of the IRefactoring interface.   The renameTarget (Declaration) and newName are the only parameters consumed by the Rename implementations (ParserState is constructor injected).  It is also important to keep in mind that the renameTarget parameter may be different than what the user had selected on the UI.  The pre-processing/conversion from the user selected target to the 'rename' target is currently the responsibility of the RenameRefactoring class which implements the IRefactoring interface.  

So, the RenameRefactoring class handles the user interactions and prepares for the operation and then delegates the task to the appropriate 'flavor' of Rename class...the Rename class' only job is to accept the target/name parameters and make-it-so.  It seems a fairly clean division of responsibility.  And, keeping the Rename classes out of the business of UI/user interactions, may allow them to more easily support operations in other/future contexts where user input does not directly initiate the rename.